presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -174,7 +174,12 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                         ImmutableSet.of(new SimplifyCountOverConstant())),
                 new WindowFilterPushDown(metadata), // This must run after PredicatePushDown and LimitPushDown so that it squashes any successive filter nodes and limits
                 new MergeWindows(),
-                new ReorderWindows(), // Should be after MergeWindows to avoid unnecessary reordering of mergeable windows
+                new IterativeOptimizer(
+                        stats,
+                        ImmutableSet.of(
```

Comment:
It was there before I removed it while rebasing here: https://github.com/prestodb/presto/pull/7300#issuecomment-292247883.
The thing is `ReorderWindows` rewriter needs `PruneIdentityProjections` to work effectively but you removed that rewriter already. That's why I decided to remove it entirely because it won't be very effective anyway.